PROTO_tests/tests/array_view_test.py

```diff
@@ -0,0 +1,203 @@
+from itertools import product
+
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+SHAPE = {
+    4: (2, 2),
+    30: (5, 3, 2),
+    252: (4, 9, 7),
+    693: (3, 7, 3, 11),
+}
+
+SIZE = [4, 30, 252, 693]
+NO_BOOL = [ak.int64, ak.float64, ak.uint64]
+
+class TestArrayView:
+    @pytest.mark.parametrize("size", SIZE)
+    @pytest.mark.parametrize("num_type", NO_BOOL)
+    def test_mulitdimensional_array_creation(self, size, num_type):
+        n = np.array([[0, 0], [0, 1], [1, 1]])
+        a = ak.array([[0, 0], [0, 1], [1, 1]])
+        assert np.array_equal(n.tolist(), a.to_list())
+        n = np.arange(size).reshape(SHAPE[size])
+        a = ak.arange(size).reshape(SHAPE[size])
+        assert np.array_equal(n.tolist(), a.to_list())
+        n = np.arange(size, dtype=num_type).reshape(SHAPE[size])
+        a = ak.arange(size, dtype=num_type).reshape(SHAPE[size])
+        assert np.array_equal(n.tolist(), a.to_list())
+
+    def test_arrayview_int_indexing(self):
+        nd = np.arange(9).reshape(3, 3)
+        pd_reshape = ak.arange(9).reshape(3, 3)
+        pd_array = ak.array([[0, 1, 2], [3, 4, 5], [6, 7, 8]])
+
+        nd_ind = [nd[i, j] for (i, j) in product(range(3), repeat=2)]
+        reshape_ind = [pd_reshape[i, j] for (i, j) in product(range(3), repeat=2)]
+        array_ind = [pd_array[i, j] for (i, j) in product(range(3), repeat=2)]
+        assert nd_ind == reshape_ind
+        assert nd_ind == array_ind
+
+        with pytest.raises(IndexError):
+            # index out bounds (>= dimension)
+            # index 3 is out of bounds for axis 0 with size 3
+            pd_reshape[3, 1]
+        with pytest.raises(IndexError):
+            # index -4 is out of bounds for axis 1 with size 3
+            pd_reshape[2, -4]
+        with pytest.raises(IndexError):
+            # too many indicies for array: array is 2-dimensional, but 3 were indexed
+            pd_reshape[0, 1, 1]
+        with pytest.raises(ValueError):
+
+            # cannot reshape array of size 9 into shape (4,3)
+            ak.arange(9).reshape(4, 3)
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_int_list_indexing(self, size):
+        N = len(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+
+        iind = ak.ones(N, dtype=ak.uint64)
+        uind = ak.cast(iind, np.uint64)
+        assert np.array_equal(iav[iind], iav[uind])
+        assert np.array_equal(uav[iind], uav[uind])
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_set_index(self, size):
+        inav = np.arange(size).reshape(SHAPE[size])
+        unav = np.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=ak.uint64).reshape(SHAPE[size])
+
+        nind = (1,) * inav.ndim
```

Comment:
Extra close review here about what values should be populating this

Reply:
Yah i think all 1's should be fine, but if we wanted to do random indexing we could do something like this
```python
import random
tuple(random.randint(0,y-1) for y in shape)
```